src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.Api/Controllers/StatelessDataController.cs

```diff
@@ -111,11 +157,27 @@ public async Task<ActionResult> Put([FromQuery] string dataType)
                 return BadRequest($"Invalid dataType {dataType} provided. Please provide a valid dataType as query parameter.");
             }
 
-            if (Request.ContentLength == null || Request.ContentLength <= 0)
+            if (GetPartyHeader(HttpContext).Count > 1)
+            {
+                return BadRequest($"Invalid party. Only one allowed");
+            }
+
+            int? partyId = await GetPartyId(HttpContext);
+
+            if (!partyId.HasValue)
+            {
+               return StatusCode((int)HttpStatusCode.Forbidden);
+            }
+
+            EnforcementResult enforcementResult = await AuthorizeAction(org, app, partyId.Value, "read");
+
+            if (!enforcementResult.Authorized)
             {
-                return BadRequest("No data found in content.");
+                return Forbidden(enforcementResult);
             }
 
+            _altinnAppContext.SetContext(new AltinnAppContext() { PartyId = partyId.Value });
```

Comment:
I don't like it that we set the context in a controller. We're forced to add it to all methods on all controllers. I'm afraid an app developer will attempt to access the context in a situation where it's not been assigned.

Reply:
Agree, but I dont see any solution how to prevent it. The controller is the only one that understand the "context" of the call

Reply:
See my comment below. The AltinnAppContext can find all it needs in the HttpContext can't it?